sdk/src/server/mod.rs

```diff
@@ -0,0 +1,242 @@
+use std::{
+    collections::hash_map::DefaultHasher,
+    env,
+    hash::{Hash, Hasher},
+    path::PathBuf,
+};
+
+use axiom_circuit::{
+    axiom_eth::{halo2curves::bn256::Fr, utils::build_utils::keygen::read_srs_from_dir},
+    run::{aggregation::agg_circuit_run, inner::run},
+    scaffold::{AxiomCircuit, AxiomCircuitScaffold},
+};
+use ethers::providers::{Http, Provider};
+use rocket::State;
+use serde::de::DeserializeOwned;
+
+use self::types::{
+    AggregationCircuitCtx, AxiomComputeCircuitCtx, AxiomComputeCtx, AxiomComputeJobStatus,
+    AxiomComputeManager, AxiomComputeServerCmd,
+};
+use crate::utils::io::{read_agg_pk_and_pinning, read_metadata, read_pinning, read_pk};
+
+pub mod types;
+
+pub async fn add_job(ctx: &State<AxiomComputeManager>, job: String) -> u64 {
+    let mut hasher = DefaultHasher::new();
+    job.hash(&mut hasher);
+    let job_id = hasher.finish();
+    if ctx.inputs.lock().unwrap().contains_key(&job_id) {
+        return job_id;
+    }
+    ctx.job_queue.lock().unwrap().push(job_id);
+    ctx.inputs.lock().unwrap().insert(job_id, job);
+    ctx.job_status
+        .lock()
+        .unwrap()
+        .insert(job_id, AxiomComputeJobStatus::Received);
+    job_id
+}
+
+pub fn prover_loop<A: AxiomCircuitScaffold<Http, Fr>, I: Into<A::InputValue> + DeserializeOwned>(
```

Comment:
@nyunyunyunyu can you take a look at this function specifically? It's for the proving server of client circuit.